### ISSUE TITLE
Evergreen Bootstrapper Installation Failure on Windows 10

### ISSUE THREAD

**Issue by developernotes** [MicrosoftEdge/WebView2Feedback#2094]
Hello,

We are attempting to install the Evergreen Bootstrapper downloaded from [here](https://go.microsoft.com/fwlink/p/?LinkId=2124703) on a Windows 10 VM. The installation fails every time, displaying the following message within the installer dialog:

<img width="760" alt="Screen Shot 2022-01-20 at 4 50 37 PM" src="https://user-images.githubusercontent.com/3225/150588568-315180f0-6512-4723-ad7d-fc0b0aa9d5e8.png">

We captured a [Process Monitor](https://docs.microsoft.com/en-us/sysinternals/downloads/procmon) output filtered on the process name `MicrosoftEdgeWebview2Setup.exe` during an installation attempt which reports an exit status of `-2147220988`. I have included the full procmon output file [here](https://zetetic.box.com/s/cvoe2hsljvgsqj6iwek5hdsu15tihuiz) for further review. This machine is up to date on Windows Updates. 

- Is there any insight as to why the Evergreen Bootstrapper continues to fail on this machine? 
- Is there a way to capture a debug log from the installer? 

Any suggestions are welcome.

Thank you,

[AB#38138768](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/38138768)

**Comment by champnic** [comment]
Hey @developernotes - How are you invoking the installer? Does it work if you just double-click on the installer yourself?

**Comment by developernotes** [comment]
Hi @champnic,

Thanks for the follow-up. We are embedding the bootstrapper within our installer which invokes the bootstrapper with the following argument: `/install`. That said, we have downloaded the bootstrapper directly onto this machine (outside of our installer) and received the same error dialog post above when double-clicking the installer directly. Is there a way to collect a log from the bootstrapper, or any other way to diagnose the cause further?

Thank you,

**Comment by champnic** [comment]
Thanks for confirming. I'll reach out to the setup team to see if there are logs that can help them diagnose here.

**Comment by champnic** [comment]
@developernotes If you could share the log in `C:\ProgramData\Microsoft\EdgeUpdate\Log\MicrosoftEdgeUpdate.log` that would be helpful. Thanks!

**Comment by champnic** [comment]
Thanks - that log doesn't show anything out of the ordinary. Would you mind also sharing this log if it exists?
`%localappdata%\Temp\MicrosoftEdgeUpdate.log`
That will include info for un-elevated installation attempts.

**Comment by developernotes** [comment]
Hi @champnic 

Please find the requested log file here:[MicrosoftEdgeUpdate.log](https://github.com/MicrosoftEdge/WebView2Feedback/files/7995151/MicrosoftEdgeUpdate.log). Thanks for looking into this!

**Comment by champnic** [comment]
Thanks! That log shows the error. Our current theories:
1. A man-in-the-middle attack is causing an unexpected file to be downloaded instead of the one you expect.
2. The user folder that the file is downloaded into has been locked down in some way. It looks like the file is getting downloaded to `%USERPROFILE%\AppData\Local\Microsoft\EdgeUpdate\Download\{F3017226-FE2A-4295-8BDF-00C3A9A7E4C5}\97.0.1072.76\MicrosoftEdge_X64_97.0.1072.76.exe`. Do you know if permissions have been changed on that folder, or could there be an antivirus meddling with the file in some way?

As a test, you can download http://msedge.f.tlu.dl.delivery.mp.microsoft.com/filestreamingservice/files/d50f46a3-4380-4aae-bb77-728aa8d1b6a1?P1=1644247347&P2=404&P3=2&P4=XYCioDSyKiQ7SPi6%2fgXogCBjU22vUxWMomTQxVeVeGXe5IU06lEnlbRSkllZyzfyHX4M%2bYVtPybfbpWPYdTIvg%3d%3d from the browser and rename it to `MicrosoftEdge_X64_97.0.1072.76.exe`.  Then you can run

`MicrosoftEdge_X64_97.0.1072.76.exe --msedgewebview --user-level` 

which should install the runtime to `%localappdata%\microsoft\EdgeWebView`.

**Comment by developernotes** [comment]
Hi @champnic 

Unfortunately, it looks like the directory doesn't exist:
<img width="737" alt="Screen Shot 2022-02-04 at 2 10 07 PM" src="https://user-images.githubusercontent.com/3225/152589674-b62483af-34aa-4a0b-890b-96f609187775.png">

We downloaded the executable you linked to and attempted the install from the command line, however, we received this error:
<img width="795" alt="Screen Shot 2022-02-04 at 2 15 08 PM" src="https://user-images.githubusercontent.com/3225/152589737-210edaae-8e80-407f-9601-1952d75f9b46.png">

Is there anything else you would like us to try? Thanks!

**Comment by developernotes** [comment]
Hi @champnic 

Just a quick update. When the executable test was just run, nothing visually happened, the command prompt just returned. Checking for the `%localappdata%\microsoft\EdgeWebView` folder is what produced the error dialog.

**Comment by champnic** [comment]
Can you resend the latest `%localappdata%\Temp\MicrosoftEdgeUpdate.log` to see if it captures the latest failure?

**Comment by developernotes** [comment]
Hi @champnic 

Here is the latest [MicrosoftEdgeUpdate.log](https://github.com/MicrosoftEdge/WebView2Feedback/files/8016448/MicrosoftEdgeUpdate.log) from `%localappdata%\Temp\`.

**Comment by champnic** [comment]
Sounds like that log is showing the same as previous ones. We aren't seeing this issue widely, so it's likely something specific about that machine. I'm not sure if an antivirus or something else is messing with the permissions of that file. Or maybe something specific about how the setup is downloading the installer?

If you do it completely manually using this link, does it install?
https://developer.microsoft.com/en-us/microsoft-edge/webview2/consumer/

**Comment by developernotes** [comment]
Hi @champnic 

Unfortunately, this caused the same error message to be displayed:

![Screen Shot 2022-02-11 at 2 00 41 PM](https://user-images.githubusercontent.com/3225/153654528-cd586751-6349-4267-8a43-6cd886c91d33.png)

Please find the latest [MicrosoftEdgeUpdate.log](https://github.com/MicrosoftEdge/WebView2Feedback/files/8051066/MicrosoftEdgeUpdate.log).

**Comment by developernotes** [comment]
Hi @champnic 

I wanted to provide a follow-up to [this](https://github.com/MicrosoftEdge/WebView2Feedback/issues/2094#issuecomment-1036531517). There is no 3rd party anti-virus software installed. While Microsoft Windows Defender was previously enabled with our install attempts, we disabled Windows Defender and received the same error dialog.

**Comment by developernotes** [comment]
Hi @champnic,

I wanted to provide an update on the issue we have been experiencing with the bootstrap installer. We were running this installer on a Windows 10 VM within VMWare Fusion. The network adapter was set to Bridged Networking → Autodetect. When we switched the adapter configuration to Internet sharing → Share with my Mac the installer completed successfully. We then uninstalled, switched back to bridged autodetect and the installer failed. Switching to Share with my Mac appears to be the solution. Thanks for your help troubleshooting this along the way!
